internal/iam/command_ipfilter_update.go

```diff
@@ -97,46 +97,46 @@ func (c *ipFilterCommand) update(cmd *cobra.Command, args []string) error {
 	}
 
 	// Create a map of the current IP group IDs
-	currentIpGroupIdsMap := make(map[string]Operation)
+	currentIpGroupIdsMap := make(map[string]bool)
```

Comment:
Feel free to take as many of the below suggestions as you have time for:
1. Please use the `pkg/set` package here!
2. If we make these warning messages a bit more generic we should be able to share this code between IP groups and IP filters (and other commands, in the future)
3. If we do that it would be great to unit test this!

Reply:
1. Oh shoot - I will look at redoing this!
2. I modeled these errors to be identical to the [new cc-policy-service responses here](https://github.com/confluentinc/cc-policy-service/pull/164). Maybe there's a way we could generalize them? 
3. Noobie question, what value does unit testing give us here if we have integration tests set up? I'm happy to do it, just trying to understand what we're looking to test here that isn't covered in the integration test responses with the `.golden`s

Reply:
> Noobie question, what value does unit testing give us here if we have integration tests set up?

Right now, this function isn't written in a way that can be unit tested, but if we write a generalized function with the following header, we would be able to unit test that function:
```
func combineAddAndRemove(add, remove []string) []string
```

Reply:
Ahh - I see. Thanks!

Reply:
(As a result, our integration tests wouldn't have to be as comprehensive. In general, unit tests should be preferred since they run much faster and it's easier to understand specifically what code they test)

Reply:
> Maybe there's a way we could generalize them?

Something like the following?
`[WARN] "%s" is marked for addition and deletion`
`[WARN] "%s" is marked for deletion but does not exist`

Reply:
I'm also realizing that if we use a map or a set, then the order of the CIDR blocks / IP group IDs are random, and so the integration tests randomly fail compared to the `.golden` output. I will have it sort these before printing.

Reply:
The newest push has the set changes a bunch more testing, as well as marking this to only be for cloud instead of cloud and on-prem.

Reply:
Fixed the sempahore tests that were failing and also made the errors more generic / reusable. Re-requesting review, thanks!